### PR TITLE
feat: 支持全局配置 CRUD

### PR DESCRIPTION
**Pull request by zouxingyuks**

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-hcm?pullRequest=1193) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-hcm?pullRequest=1193) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: chenjr15<br/>:white_check_mark: fxdgg<br/>:white_check_mark: ZQHcode<br/>:x: zouxingyuks<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/bk-hcm?pullRequest=1193) it.</sub>

File: pkg/api/data-service/global_config/global_config.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+	return nil
+}
+
+// ListResp ...
+type ListResp core.ListResultT[tablegconf.GlobalConfigTable]
+
+// BatchCreateReq ...
+type BatchCreateReq struct {
+	Configs []GlobalConfig `json:"configs" validate:"required,min=1"`
```
Comment (yangfan-njupt) [inline pkg/api/data-service/global_config/global_config.go:54]: max=100

File: pkg/api/data-service/global_config/global_config.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	if err := validator.Validate.Struct(req); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// BatchUpdateReq ...
+type BatchUpdateReq struct {
+	Configs []GlobalConfig `json:"configs" validate:"required,min=1"`
```
Comment (yangfan-njupt) [inline pkg/api/data-service/global_config/global_config.go:68]: 同上，max=100

File: pkg/dal/table/global-config/global_config.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	Creator string `db:"creator" json:"creator"`
+	// Reviser 更新者
+	Reviser string `db:"reviser" json:"reviser"`
+	// CreatedAt 创建时间
+	CreatedAt types.Time `db:"created_at" json:"created_at"`
+	// UpdatedAt 更新时间
+	UpdatedAt types.Time `db:"updated_at" json:"updated_at"`
+}
+
+// UniqueKey returns the unique key of BizOrgRel table.
```
Comment (yangfan-njupt) [inline pkg/dal/table/global-config/global_config.go:72]: BizOrgRel table 名字不对哈

File: pkg/tools/util/conv.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	"fmt"
+)
+
+// GetStrByInterface interface to string
+func GetStrByInterface(a interface{}) string {
+	if nil == a {
+		return ""
+	}
+	return fmt.Sprintf("%v", a)
```
Comment (yangfan-njupt) [inline pkg/tools/util/conv.go:25]: 这个真的符合预期么？

File: scripts/sql/9999_20241114_1100_global_config.sql [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+create table if not exists `global_config`
+(
+    `id`           varchar(64) not null comment '主键',
+    `config_key`   varchar(64) not null comment 'key',
+    `config_value` json        not null comment 'value',
+    `config_type`  varchar(64) not null comment '类型',
+    `memo`         varchar(255)         default '' comment '备注',
+    `creator`      varchar(64) not null comment '创建者',
+    `reviser`      varchar(64) not null comment '更新者',
+    `result`       json                 default NULL comment '结果',
```
Comment (yangfan-njupt) [inline scripts/sql/9999_20241114_1100_global_config.sql:39]: result这一列好像代码里没体现，预留用于什么用途？

File: scripts/sql/9999_20241114_1100_global_config.sql [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    `created_at`   timestamp   not null default current_timestamp comment '创建时间',
+    `updated_at`   timestamp   not null default current_timestamp on update current_timestamp comment '更新时间',
+    primary key (`id`),
+    unique key `idx_global_config_id` (`config_type`, `config_key`)
```
Comment (yangfan-njupt) [inline scripts/sql/9999_20241114_1100_global_config.sql:43]: 考虑加个config_type + config_key的索引吧

File: pkg/api/data-service/global_config/global_config.go [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+// Validate FindReq
+func (req *FindReq) Validate() error {
+	if err := validator.Validate.Struct(req); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GlobalConfig define cvm table.
```
Comment (chenjr15) [inline pkg/api/data-service/global_config/global_config.go:119]: 注释错误
这个结构体应该定义在pkg/api/core里

File: pkg/api/data-service/global_config/global_config.go [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+type GlobalConfig struct {
+	// ID global config id
+	ID string `json:"id"`
+	// ConfigKey global config key, key+type is unique
+	ConfigKey string `json:"config_key"`
+	// ConfigValue global config value, json format
+	ConfigValue interface{} `json:"config_value"`
```
Comment (chenjr15) [inline pkg/api/data-service/global_config/global_config.go:126]: 这个不要用interface{} 用json.RawMessage，不要去转换数据

File: cmd/data-service/service/global-config/global_config.go [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+	}
+
+	// create
+	ids, err := svc.dao.Txn().AutoTxn(cts.Kit, func(txn *sqlx.Tx, opt *orm.TxnOption) (interface{}, error) {
+		globalConfigs := make([]tablegconf.GlobalConfigTable, len(req.Configs))
+		for index, config := range req.Configs {
+			globalConfigs[index] = tablegconf.GlobalConfigTable{
+				ConfigKey:   config.ConfigKey,
+				ConfigValue: dtypes.JsonField(util.GetStrByInterface(config.ConfigValue)),
+				ConfigType:  config.ConfigType,
```
Comment (chenjr15) [inline cmd/data-service/service/global-config/global_config.go:61]: ConfigValue create 和 list 都用json.RawMessage, 不要去转换格式

File: pkg/dal/table/global-config/global_config.go (hunk 3 above)
Comment (zouxingyuks) [reply pkg/dal/table/global-config/global_config.go:72]: 本修改将在https://github.com/TencentBlueKing/bk-hcm/pull/1215 中修正

File: pkg/api/data-service/global_config/global_config.go (hunk 7 above)
Comment (zouxingyuks) [reply pkg/api/data-service/global_config/global_config.go:119]: 本修改将在https://github.com/TencentBlueKing/bk-hcm/pull/1215 中修正

File: pkg/api/data-service/global_config/global_config.go (hunk 8 above)
Comment (zouxingyuks) [reply pkg/api/data-service/global_config/global_config.go:126]: 本修改将在https://github.com/TencentBlueKing/bk-hcm/pull/1215 中修正

File: pkg/tools/util/conv.go (hunk 4 above)
Comment (zouxingyuks) [reply pkg/tools/util/conv.go:25]: 本修改将在https://github.com/TencentBlueKing/bk-hcm/pull/1215 中修正

File: pkg/api/data-service/global_config/global_config.go (hunk 2 above)
Comment (zouxingyuks) [reply pkg/api/data-service/global_config/global_config.go:68]: 本修改将在https://github.com/TencentBlueKing/bk-hcm/pull/1215 中修正

**Review by chenjr15 — APPROVED** [review]
当前PR先合入，后续pr修复问题

**Pull request merged**